mesonbuild/msubprojects.py

```diff
@@ -435,6 +436,33 @@ def update(self) -> bool:
             self.wrap.update_hash_cache(self.repo_dir)
         return success
 
+    def syncwrap(self) -> bool:
+        if self.wrap.type != 'git' and not self.wrap.redirected:
+            return True
+
+        self.log(f'Sync {self.wrap.name}...')
+        if self.wrap.redirected:
+            self.wrap.filename = Path(self.wrap.original_filename).resolve()
+            self.wrap.parse_wrap()
+
+        if self.wrap.type != 'git':
+            return True
+
+        revision = self.wrap.get('revision').strip()
+        if revision.lower() == 'head':
+            return True
+
+        new_revision = self.git_output(['rev-parse', 'HEAD']).strip()
```

Comment:
Wondering if we could use tag name or branch name, maybe optionally, instead of commit hash.

Reply:
It's not really my use case so i'm not entirely sure how that would work but i added a `--tags` option which updates the wrap only if the repository matches a clean tag.